skglm/estimators.py

```diff
@@ -1,31 +1,201 @@
-from sklearn.linear_model import ElasticNet as ElasticNet_sklearn
-from sklearn.linear_model import LogisticRegression as LogReg_sklearn
+from sklearn.utils.validation import check_is_fitted
+from sklearn.utils import check_array, check_consistent_length
 from sklearn.linear_model import MultiTaskLasso as MultiTaskLasso_sklearn
-from sklearn.linear_model._base import LinearModel
-from sklearn.svm import LinearSVC as LinearSVC_sklearn
+from sklearn.linear_model._base import (
+    _preprocess_data, LinearModel, RegressorMixin,
+    LinearClassifierMixin, SparseCoefMixin, BaseEstimator
+)
+from sklearn.utils.extmath import softmax
 from sklearn.preprocessing import LabelEncoder
-from sklearn.multiclass import OneVsRestClassifier
+from sklearn.multiclass import OneVsRestClassifier, check_classification_targets
 
-from sklearn.linear_model._base import _preprocess_data
 
-from skglm.penalties import (
-    L1, WeightedL1, L1_plus_L2, MCPenalty, IndicatorBox, L2_1
-)
-from skglm.datafits import (
-    Quadratic, Logistic, QuadraticSVC, QuadraticMultiTask
-)
 from skglm.utils import compiled_clone
-from skglm.solvers import cd_solver_path, bcd_solver_path
+from skglm.solvers import cd_solver_path, multitask_bcd_solver_path
+from skglm.solvers.cd_solver import cd_solver
+from skglm.solvers.multitask_bcd_solver import multitask_bcd_solver
+from skglm.datafits import Quadratic, Logistic, QuadraticSVC, QuadraticMultiTask
+from skglm.penalties import L1, WeightedL1, L1_plus_L2, MCPenalty, IndicatorBox, L2_1
+
+
+def _glm_reg_fit(X, y, model, datafit, penalty):
+    check_X_params = dict(
+        dtype=[np.float64, np.float32], order='F',
+        accept_sparse='csc', copy=model.fit_intercept)
+    check_y_params = dict(ensure_2d=False, order='F')
+
+    X, y = model._validate_data(
+        X, y, validate_separately=(check_X_params, check_y_params))
+    X = check_array(X, 'csc', dtype=[np.float64, np.float32],
+                    order='F', copy=False, accept_large_sparse=False)
+    y = check_array(y, 'csc', dtype=X.dtype.type, order='F', copy=False,
+                    ensure_2d=False)
+
+    if y.ndim == 2 and y.shape[1] == 1:
+        warnings.warn("DataConversionWarning('A column-vector y"
+                      " was passed when a 1d array was expected")
+        y = y[:, 0]
+
+    if not hasattr(model, "n_features_in_"):
+        model.n_features_in_ = X.shape[1]
+
+    n_samples = X.shape[0]
+    if n_samples != y.shape[0]:
+        raise ValueError("X and y have inconsistent dimensions (%d != %d)"
+                         % (n_samples, y.shape[0]))
+
+    if not model.warm_start or not hasattr(model, "coef_"):
+        model.coef_ = None
+
+    penalty = compiled_clone(penalty)
+    datafit = compiled_clone(datafit, to_float32=X.dtype == np.float32)
+    if issparse(X):
+        datafit.initialize_sparse(X.data, X.indptr, X.indices, y)
+    else:
+        datafit.initialize(X, y)
+
+    X_ = X
```

Comment:
This is not needed in this function, it's reminiscent from glm_classif. It could be confusing. I would merge the two functions back